### PR TITLE
Implements the validation operators to the types DateOnly and DateTime objects.

### PR DESCRIPTION
**Pull request by PhilipeRLeal**

### CODE REVIEW

File: SpreadCheetah/SpreadCheetah.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;net6.0;net7.0;net8.0</TargetFrameworks>
```
Comment (sveinungf) [inline SpreadCheetah/SpreadCheetah.csproj:4]: We should not drop support for .NET Standard 2.0. I assume you removed it because of `DateOnly`, which is only available on .NET 6 and later. Using `DateOnly` would be nice for this feature, but I don't think it is enough to drop the backwards compatibility. So for now we should basically only use `DateTime` for this feature.

File: SpreadCheetah.Test/Tests/SpreadsheetDataValidationTests.cs [hunk 2]
```diff
@@ -11,6 +11,68 @@ namespace SpreadCheetah.Test.Tests;
...
+        // Assert
+        SpreadsheetAssert.Valid(stream);
+        using var workbook = new XLWorkbook(stream);
+        var worksheet = workbook.Worksheets.Single();
+        var actualValidation = Assert.Single(worksheet.DataValidations);
+        var actualRange = Assert.Single(actualValidation.Ranges);
+        var cell = Assert.Single(actualRange.Cells());
+        Assert.Equal(1, cell.Address.ColumnNumber);
+        Assert.Equal(1, cell.Address.RowNumber);
+        Assert.Equal(XLAllowedValues.AnyValue, actualValidation.AllowedValues);
```
Comment (sveinungf) [inline SpreadCheetah.Test/Tests/SpreadsheetDataValidationTests.cs:69]: I would expect `XLAllowedValues.Date` here, not `XLAllowedValues.AnyValue`.

File: SpreadCheetah/Validations/DataValidation.cs [hunk 3]
```diff
@@ -74,6 +74,64 @@ public ValidationErrorType ErrorType
...
+
+    #endregion DateOnly validations
+
+    #region DateTime validations
+    private static DataValidation _DateTime(ValidationOperator op, DateTime min, DateTime max)
+    {
+        if (max < min)
+            throw new ArgumentException(MinGreaterThanMaxMessage, nameof(min));
+        else
+            return new DataValidation(ValidationType.DateTime, op, min.ToStringInvariant(), max.ToStringInvariant());
```
Comment (sveinungf) [inline SpreadCheetah/Validations/DataValidation.cs:109]: Are you sure about storing `DateTime` values as strings like this is correct for the Excel file?

File: SpreadCheetah/Validations/ValidationType.cs [hunk 4]
```diff
@@ -6,5 +6,7 @@ internal enum ValidationType
     Integer,
     Decimal,
     TextLength,
-    List
+    List,
+    DateOnly,
+    DateTime
```
Comment (sveinungf) [inline SpreadCheetah/Validations/ValidationType.cs:11]: You need to also update `DataValidationXml` to handle the enum value.

**Review by sveinungf — CHANGES_REQUESTED** [review]

File: SpreadCheetah/SpreadCheetah.csproj (hunk 1 above)
Comment (PhilipeRLeal) [reply SpreadCheetah/SpreadCheetah.csproj:4]: No problems. I will revert compatibility in the next commit.

File: SpreadCheetah.Test/Tests/SpreadsheetDataValidationTests.cs (hunk 2 above)
Comment (PhilipeRLeal) [reply SpreadCheetah.Test/Tests/SpreadsheetDataValidationTests.cs:69]: I could not find where this property is being set; would you be so kind as to point me the right file/class responsible for this "AllowedValues" property setting?

File: SpreadCheetah/Validations/ValidationType.cs (hunk 4 above)
Comment (PhilipeRLeal) [reply SpreadCheetah/Validations/ValidationType.cs:11]: Could you be more specific as to where and how I should update DataValidationXml?

Comment (PhilipeRLeal) [reply SpreadCheetah/Validations/ValidationType.cs:11]: Would it be the `TryWriteType `method that needs to be updated?

Comment (sveinungf) [reply SpreadCheetah/Validations/ValidationType.cs:11]: Yes 🙂

File: SpreadCheetah.Test/Tests/SpreadsheetDataValidationTests.cs (hunk 2 above)
Comment (sveinungf) [reply SpreadCheetah.Test/Tests/SpreadsheetDataValidationTests.cs:69]: I think this will be set correctly once you've made the required changes in `DataValidationXml` (related to the other comment).

File: SpreadCheetah/Validations/ValidationType.cs (hunk 4 above)
Comment (PhilipeRLeal) [reply SpreadCheetah/Validations/ValidationType.cs:11]: Do you have any documentation that explains the different data types (and respective strings) supported by the method? I have tried the following options:

1. `ValidationType.DateTime => "type=\"datetime\" "u8.TryCopyTo(bytes, ref bytesWritten),`
2. `ValidationType.DateTime => "type=\"date\" "u8.TryCopyTo(bytes, ref bytesWritten),`


So far, it seems that the latter is the correct choice. Nevertheless, I get uneasy do to its name which implies solely in a DateOnly object; therefore, losing the time property of the Datetime element.

Comment (sveinungf) [reply SpreadCheetah/Validations/ValidationType.cs:10]: The documentation is basically the Open XML specification, but it can be a bit hard to read. So I suggest you instead:
1. Create the validation in Excel
2. Save it to an `.xlsx` file
3. Rename the `.xlsx` file to `.zip`
4. Unzip that file
5. Look at the content of the `xl/worksheets/sheet1.xml` file

Comment (PhilipeRLeal) [reply SpreadCheetah/Validations/ValidationType.cs:10]: Dear Sveinungf, thank you for the feedback. As it seems, after unzip, within the docProps/core.xml file, I managed to retrieve the following type: dcterms:W3CDTF. I am not sure whether or not this is the correct type. If we look at the ./xl/worksheets/sheet1.xml file, we can find the "date" type. Therefore, so far, I would keep the "date" type, instead of the "dcterms:W3CDTF".

File: SpreadCheetah/Validations/DataValidation.cs (hunk 3 above)
Comment (PhilipeRLeal) [reply SpreadCheetah/Validations/DataValidation.cs:83]: Regarding datetime being represented as string, Excel follows the Culture in which it is installed. Therefore, for north american people, the datetime string representation would be like the following: "yyyy/mm/dd HH:MM:ss:ms"; for latin people, the same datetime would be represented in the following format: "dd/mm/yyyy HH:MM:ss:ms".

Regardless of how datetime is string represented, within Excel, a datetime is basically an integer (i.e., ticks since EPOCH). Therefore, one could also define datetime as an integer number.

File: SpreadCheetah/Validations/ValidationType.cs (hunk 4 above)
Comment (PhilipeRLeal) [reply SpreadCheetah/Validations/ValidationType.cs:10]: After some trials, the `ValidationType.DateTime => "type=\"date\" "u8.TryCopyTo(bytes, ref bytesWritten),` option was the right choice. 

All tests are passing.

File: SpreadCheetah/Validations/DataValidation.cs (hunk 3 above)
Comment (PhilipeRLeal) [reply SpreadCheetah/Validations/DataValidation.cs]: Dear Sveinungf, after some digging, I found the standard definition of date for Excel. See reference [here](https://support.microsoft.com/en-us/office/date-systems-in-excel-e7fe7167-48a9-4b96-bb53-5612a800b487). 

In this last commit, the Datetime representation has been adjusted to the provided reference format.

Sincerely,

**Comment by PhilipeRLeal** [comment]
Dear Sveinungf, please let me know if you need any further adjustments to the provided Pull Request.

Sincerely,

File: SpreadCheetah/Validations/DataValidation.cs [hunk 5]
```diff
@@ -74,6 +75,50 @@ public ValidationErrorType ErrorType
...
+            var minElapsedDates = (min.AddDays(2) - epoch).Days;
+            var maxElapsedDates = (max.AddDays(2) - epoch).Days;
+
+            return new DataValidation(ValidationType.DateTime, op, minElapsedDates.ToString(CultureInfo.InvariantCulture), maxElapsedDates.ToString(CultureInfo.InvariantCulture));
+        }
+    }
+
+    private static DataValidation _DateTime(ValidationOperator op, DateTime value)
+    {
+        var epoch = new DateTime(1900,1,1,1,1,1,DateTimeKind.Unspecified);
```
Comment (sveinungf) [inline SpreadCheetah/Validations/DataValidation.cs:95]: Maybe we can use `DateTime.ToOADate`? https://learn.microsoft.com/en-us/dotnet/api/system.datetime.tooadate

File: SpreadCheetah.SourceGenerator.Test/SpreadCheetah.SourceGenerator.Test.csproj [hunk 6]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net472;net8.0;net9.0</TargetFrameworks>
+    <TargetFrameworks>net6.0;net8.0;net9.0</TargetFrameworks>
```
Comment (sveinungf) [inline SpreadCheetah.SourceGenerator.Test/SpreadCheetah.SourceGenerator.Test.csproj:4]: Can you revert this change?

File: SpreadCheetah.Test/Helpers/DataValidationGenerator.cs [hunk 7]
```diff
@@ -23,6 +23,16 @@ public static IList<DataValidation> Generate(int count)
 
     private static readonly Func<Faker, DataValidation>[] Factories =
     [
+        f => DataValidation.DateTimeBetween(f.Random.SimpleDateTimePair(out var max), max),
+        f => DataValidation.DateTimeEqualTo(f.Random.SimpleDateTime()),
```
Comment (sveinungf) [inline SpreadCheetah.Test/Helpers/DataValidationGenerator.cs:27]: Maybe we can use the built-in functionality in Bogus to generate DateTimes? For example `f.Date.PastOffset()`.

File: SpreadCheetah.Test/Tests/PublicApiTests.PublicApi_Generate.DotNet6_0.verified.txt [hunk 8]
```diff
@@ -0,0 +1 @@
+﻿
```
Comment (sveinungf) [inline SpreadCheetah.Test/Tests/PublicApiTests.PublicApi_Generate.DotNet6_0.verified.txt:1]: This file should be removed.

**Review by sveinungf — CHANGES_REQUESTED** [review]

File: SpreadCheetah/Validations/DataValidation.cs (hunk 5 above)
Comment (PhilipeRLeal) [reply SpreadCheetah/Validations/DataValidation.cs:95]: We can certainly use OADate format. Bu what would we gain by using that format?

File: SpreadCheetah.SourceGenerator.Test/SpreadCheetah.SourceGenerator.Test.csproj (hunk 6 above)
Comment (PhilipeRLeal) [reply SpreadCheetah.SourceGenerator.Test/SpreadCheetah.SourceGenerator.Test.csproj]: Solved.

File: SpreadCheetah.Test/Helpers/DataValidationGenerator.cs (hunk 7 above)
Comment (PhilipeRLeal) [reply SpreadCheetah.Test/Helpers/DataValidationGenerator.cs:27]: Can you show me an example?

File: SpreadCheetah.Test/Tests/PublicApiTests.PublicApi_Generate.DotNet6_0.verified.txt (hunk 8 above)
Comment (PhilipeRLeal) [reply SpreadCheetah.Test/Tests/PublicApiTests.PublicApi_Generate.DotNet6_0.verified.txt]: Done.

File: SpreadCheetah/Validations/DataValidation.cs (hunk 5 above)
Comment (sveinungf) [reply SpreadCheetah/Validations/DataValidation.cs:95]: You have almost implemented what `DateTime.ToOADate()` does, so you can replace this:
```csharp
var epoch = new DateTime(1900,1,1,1,1,1,DateTimeKind.Unspecified);
var elapsedDate = value.AddDays(2) - epoch;
var days = (elapsedDate.Days);
```
with
```csharp
var days = value.ToOADate();
```

File: SpreadCheetah.Test/Helpers/DataValidationGenerator.cs (hunk 7 above)
Comment (sveinungf) [reply SpreadCheetah.Test/Helpers/DataValidationGenerator.cs:27]: `f => DataValidation.DateTimeEqualTo(f.Date.PastOffset())`

**Review by PhilipeRLeal — COMMENTED** [review]
This new commit solves the two requests provided by the reviewer.

**Comment by PhilipeRLeal** [comment]
Dear Sveinungf, please let me know if you need any further adjustments to this Pull Request. 

Sincerely,

**Comment by sveinungf** [comment]
Thank you for the contribution. I will make a few adjustments in a separate branch before merging it into main.
